### PR TITLE
feat: Add HOWTO: Auto HTTP-to-HTTPS for Octavia LB

### PR DESCRIPTION
**Pull request by colder-is-better**
We show how to add automatic HTTP to HTTPS redirection to an Octavia HTTPS-terminated load balancer.

### CODE REVIEW

File: docs/howto/openstack/octavia/lbaas-l7pol.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+```bash
+openstack loadbalancer listener create \
+    --name mylb-listener-http \
+    --protocol HTTP \
+    --protocol-port 80 \
+    mylb
+```
+
+```plain
```
[the remark this answers is not in the document]
Comment (colder-is-better) [reply docs/howto/openstack/octavia/lbaas-l7pol.md:107]: I prefer `console`-style code blocks muhself, but I followed the sub-optimal approach of two-fenced code blocks in an attempt to make the lives of those who would like to copy/paste commands easier :/

File: docs/howto/openstack/octavia/lbaas-l7pol.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+```
+
+## Adding policies and rules
+
+Create `mylb-listener-http-policy`, a policy for `mylb-listener-http`:
+
+```bash
+openstack loadbalancer l7policy create \
+    --action REDIRECT_PREFIX \
+    --redirect-prefix https://whoogle.parabing.com/ \
```
[the remark this answers is not in the document]
Comment (colder-is-better) [reply docs/howto/openstack/octavia/lbaas-l7pol.md:188]: Well, in the example I use the trailing `/` does not cause any issues and the redirection works as expected. But everything is fine *without* the trailing `/`, so I am going to remove it.

File: docs/howto/openstack/octavia/lbaas-l7pol.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+```plain
++---------------------------+---------------------+
+| name                      | provisioning_status |
++---------------------------+---------------------+
+| mylb-listener-http-policy | ACTIVE              |
++---------------------------+---------------------+
+```
+
+From now on, all client attempts to reach `http://whoogle.parabing.com`
+will end up at `https://whoogle.parabing.com`.
```
[the remark this answers is not in the document]
Comment (colder-is-better) [reply docs/howto/openstack/octavia/lbaas-l7pol.md:277]: Ah, yes, I am adding a redirection demo with `curl`!

File: docs/howto/openstack/octavia/lbaas-l7pol.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+| mylb-listener-http-policy | ACTIVE              |
++---------------------------+---------------------+
+```
+
+Then, add a single rule to `mylb-listener-http-policy`:
+
+```bash
+openstack loadbalancer l7rule create \
+    --compare-type STARTS_WITH \
+    --type PATH \
```
[the remark this answers is not in the document]
Comment (colder-is-better) [reply docs/howto/openstack/octavia/lbaas-l7pol.md:237]: Thanks, I modified the rule according to your suggestion -- and the redirect still works as expected!

File: docs/howto/openstack/octavia/lbaas-l7pol.md [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+| 95414884-   | 3a683bfb-6c77-  | mylb-       | dfc70046739 | TERMINATED_ |           443 | True           |
+| f9f2-4f76-  | 4ffd-befa-      | listener-   | 6428bacba43 | HTTPS       |               |                |
+| 8aa5-       | 9f5e0e168538    | https       | 76e72cc3e9  |             |               |                |
+| 56741f12825 |                 |             |             |             |               |                |
+| 2           |                 |             |             |             |               |                |
++-------------+-----------------+-------------+-------------+-------------+---------------+----------------+
+```
+
+Let us now add `mylb-listener-http`, a new HTTP-based listener for
+`mylb`:
```
[the remark this answers is not in the document]
Comment (colder-is-better) [reply docs/howto/openstack/octavia/lbaas-l7pol.md:97]: Nice catch!

**Review by fghaas — COMMENTED** [review]
I think this is nearly good to go now, I just have a few remaining suggestions:

* Use a shorter page title (because the currently proposed one is a monster :) )
* Use filename references for links within the same directory
* Replace the `parabing.com` domain with `example.com` to clarify that it's... well, an example.

I've taken the liberty to wrap those changes into aac1f8e4c2cfdcb3aea03dc50413455a555075f7; if you're good with them we can squash that commit and then merge the whole thing.

**Comment by colder-is-better** [comment]
I am fine with all remaining suggestions (and I particularly enjoyed the title "transformation", to put it mildly :))

**Review by fghaas — APPROVED** [review]

**Pull request merged**